app/views/form/page.html.erb

```diff
@@ -1,7 +1,11 @@
 <% content_for :title, @page.header.presence || @page.questions.first.header.html_safe %>
 
 <% content_for :before_content do %>
-  <%= govuk_back_link(href: "javascript:history.back()") %>
+  <% if url_for(:back).include?("check_answers") %>
+    <%= govuk_back_link(href: :back) %>
+  <% else %>
+    <%= govuk_back_link(href: send(@log.form.previous_page_redirect_path(@page, @log, current_user), @log)) %>
```

Comment:
I wonder if there is some clever way that we can use url_for() here
https://api.rubyonrails.org/v5.2/classes/ActionDispatch/Routing/UrlFor.html#method-i-url_for

Reply:
I don't believe it is possible.

Reply:
Two questions! Did you know that `:back` returns HTTP referrer or `javascript:history.back() ` if it's not set? Why do you need to use `send` here?